File: app/src/main/java/ru/geekbrains/lessions2345/notepadonfragments_2/observe/Observer.java
package ru.geekbrains.lessions2345.notepadonfragments_2.observe;

import ru.geekbrains.lessions2345.notepadonfragments_2.model.DeleteAnswersTypes;

public interface Observer {
    void completeGoogleAuthorise(boolean completeGoogleAuthorise);

    void updateDatesFromFireBase(int numberDownloadedNotes);

    void updateUserChooseDeleteFile(DeleteAnswersTypes userDecision);

    void updateUserChooseDeleteFileFromContextMenu(DeleteAnswersTypes userDecision, int indexChoosedNoteInContextMenu);
}

File: app/src/main/java/ru/geekbrains/lessions2345/notepadonfragments_2/observe/Observer.kt
package ru.geekbrains.lessions2345.notepadonfragments_2.observe

import ru.geekbrains.lessions2345.notepadonfragments_2.model.DeleteAnswersTypes

interface Observer {
    fun completeGoogleAuthorise(completeGoogleAuthorise : Boolean)

    fun updateDatesFromFireBase(numberDownloadedNotes : Int)

    fun updateUserChooseDeleteFile(userDecision : DeleteAnswersTypes)

    fun updateUserChooseDeleteFileFromContextMenu(userDecision : DeleteAnswersTypes, indexChoosedNoteInContextMenu : Int)
}